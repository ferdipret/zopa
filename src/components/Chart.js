import React from 'react'
import styled from 'styled-components'
import Chart from 'react-minimal-pie-chart'
import color from './style-utils/colors'
import { formatCurrency } from './helpers/format-currency'

function ChartWrapper({ sent, available }) {
  return (
    sent &&
    available && (
      <ChartContainer>
        <div>
          <ValueWrapper>{formatCurrency(sent)}</ValueWrapper>
          <div>total sent</div>
        </div>
        <Chart
          data={[
            { value: sent, color: color.chartStrokeSecondary() },
            { value: available, color: color.chartStrokePrimary() },
          ]}
          lineWidth={48}
          style={{
            height: '64px',
            width: '64px',
            margin: '0 30px',
          }}
          startAngle={-90}
          lengthAngle={-360}
        />
        <div>
          <ValueWrapper>{formatCurrency(available)}</ValueWrapper>
          <div>left available</div>
        </div>
      </ChartContainer>
    )
  )
}

const ChartContainer = styled.div`
  align-items: center;
  display: flex;
  font-family: Helvetica;
  justify-content: center;
  margin-bottom: 77px;
`

const ValueWrapper = styled.div`
  font-size: 16px;
  line-height: 24px;
`

export default ChartWrapper
