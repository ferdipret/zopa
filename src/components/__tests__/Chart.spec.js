import React from "react"
import { shallow } from "enzyme"
import Chart from "../Chart"

describe("Chart", () => {
  const wrapper = shallow(<Chart sent={100} available={100} />)
  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
