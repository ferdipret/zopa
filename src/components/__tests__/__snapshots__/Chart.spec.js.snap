// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChartWrapper
    available={100}
    sent={100}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <styled.div>
                    £100.00
          </styled.div>
          <div>
                    total sent
          </div>
</div>,
        <ReactMinimalPieChart
          animate={false}
          animationDuration={500}
          animationEasing="ease-out"
          cx={50}
          cy={50}
          data={
                    Array [
                              Object {
                                "color": "hsl(240, 11%, 93%)",
                                "value": 100,
                              },
                              Object {
                                "color": "hsl(39, 100%, 58%)",
                                "value": 100,
                              },
                            ]
          }
          lengthAngle={-360}
          lineWidth={48}
          paddingAngle={0}
          radius={50}
          ratio={1}
          rounded={false}
          startAngle={-90}
          style={
                    Object {
                              "height": "64px",
                              "margin": "0 30px",
                              "width": "64px",
                            }
          }
/>,
        <div>
          <styled.div>
                    £100.00
          </styled.div>
          <div>
                    left available
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <styled.div>
              £100.00
</styled.div>,
            <div>
              total sent
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "£100.00",
            },
            "ref": null,
            "rendered": "£100.00",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "total sent",
            },
            "ref": null,
            "rendered": "total sent",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animate": false,
          "animationDuration": 500,
          "animationEasing": "ease-out",
          "cx": 50,
          "cy": 50,
          "data": Array [
            Object {
              "color": "hsl(240, 11%, 93%)",
              "value": 100,
            },
            Object {
              "color": "hsl(39, 100%, 58%)",
              "value": 100,
            },
          ],
          "lengthAngle": -360,
          "lineWidth": 48,
          "paddingAngle": 0,
          "radius": 50,
          "ratio": 1,
          "rounded": false,
          "startAngle": -90,
          "style": Object {
            "height": "64px",
            "margin": "0 30px",
            "width": "64px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <styled.div>
              £100.00
</styled.div>,
            <div>
              left available
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "£100.00",
            },
            "ref": null,
            "rendered": "£100.00",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "left available",
            },
            "ref": null,
            "rendered": "left available",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <styled.div>
                        £100.00
            </styled.div>
            <div>
                        total sent
            </div>
</div>,
          <ReactMinimalPieChart
            animate={false}
            animationDuration={500}
            animationEasing="ease-out"
            cx={50}
            cy={50}
            data={
                        Array [
                                    Object {
                                      "color": "hsl(240, 11%, 93%)",
                                      "value": 100,
                                    },
                                    Object {
                                      "color": "hsl(39, 100%, 58%)",
                                      "value": 100,
                                    },
                                  ]
            }
            lengthAngle={-360}
            lineWidth={48}
            paddingAngle={0}
            radius={50}
            ratio={1}
            rounded={false}
            startAngle={-90}
            style={
                        Object {
                                    "height": "64px",
                                    "margin": "0 30px",
                                    "width": "64px",
                                  }
            }
/>,
          <div>
            <styled.div>
                        £100.00
            </styled.div>
            <div>
                        left available
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <styled.div>
                £100.00
</styled.div>,
              <div>
                total sent
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "£100.00",
              },
              "ref": null,
              "rendered": "£100.00",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "total sent",
              },
              "ref": null,
              "rendered": "total sent",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animate": false,
            "animationDuration": 500,
            "animationEasing": "ease-out",
            "cx": 50,
            "cy": 50,
            "data": Array [
              Object {
                "color": "hsl(240, 11%, 93%)",
                "value": 100,
              },
              Object {
                "color": "hsl(39, 100%, 58%)",
                "value": 100,
              },
            ],
            "lengthAngle": -360,
            "lineWidth": 48,
            "paddingAngle": 0,
            "radius": 50,
            "ratio": 1,
            "rounded": false,
            "startAngle": -90,
            "style": Object {
              "height": "64px",
              "margin": "0 30px",
              "width": "64px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <styled.div>
                £100.00
</styled.div>,
              <div>
                left available
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "£100.00",
              },
              "ref": null,
              "rendered": "£100.00",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "left available",
              },
              "ref": null,
              "rendered": "left available",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
